lib/compiler/src/beam_doc.erl

```diff
@@ -0,0 +1,241 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2023-2028. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+%% Purpose : Generate documentation as per EEP-48
+
+
+-module(beam_doc).
+
+-feature(maybe_expr, enable).
+
+-moduledoc "
```

Comment:
There should not be any `-moduledoc` for modules that do not expose a public API.

If you want some comments here they should be comments.

Elixir has the convention that you should put `-moduledoc false`, maybe we should do the same? https://hexdocs.pm/elixir/1.15.6/writing-documentation.html#hiding-internal-modules-and-functions

Reply:
Ok, I thought we use`-moduledoc` regardless of whether the module exposes a public API or not. Got it